src/librustc_resolve/late/diagnostics.rs

```diff
@@ -256,6 +261,20 @@ impl<'a> LateResolutionVisitor<'a, '_> {
         if !levenshtein_worked {
             err.span_label(base_span, fallback_label);
             self.type_ascription_suggestion(&mut err, base_span);
+            if let Some(span) = self.current_let_binding.and_then(|(pat_sp, ty_sp)| {
+                if ty_sp.contains(base_span) && is_local {
```

Comment:
Could use `.then_with(...)` here (https://github.com/rust-lang/rust/pull/65195).

Reply:
Changed it in a different way.